.gitlab/docker/Dockerfile.spack_base

```diff
@@ -0,0 +1,47 @@
+ARG BASE_IMAGE
+FROM $BASE_IMAGE
+
+ARG SPACK_REPO=https://github.com/spack/spack
+ARG SPACK_VERSION
+
+ENV DEBIAN_FRONTEND=noninteractive
+ENV SPACK_ROOT=/opt/spack
+
+# Install utility packages
+RUN apt-get update && apt-get -yqq install --no-install-recommends \
+    build-essential \
+    bzip2 \
+    ca-certificates \
+    coreutils \
+    cmake \
+    curl \
+    diffutils \
+    findutils \
+    g++ \
+    gcc \
+    git \
+    gnupg2 \
+    libtool \
+    locales \
+    m4 \
+    openssl \
+    pkgconf \
+    python3 \
+    python3-pip \
+    tar \
+    unzip \
+    vim \
+    xz-utils \
+  && rm -Rf /var/lib/apt/lists/*
+
+# Install spack
+RUN mkdir -p $SPACK_ROOT \
+    && git clone --depth 1 -b ${SPACK_VERSION} ${SPACK_REPO} ${SPACK_ROOT}
+
+ENV PATH $SPACK_ROOT/bin:$PATH
+
+# Install utility
+RUN spack external find --scope site
+RUN spack compiler find --scope site
+
+ENV CMAKE_GENERATOR=Ninja
```

Comment:
Does this need to be in the base image? Or does it need to be set at all? We should definitely use ninja for building pika but it might be enough to use `-GNinja`?